File: solutions/typescript/armstrong-numbers/3/armstrong-numbers.ts
class ArmstrongNumbers {

  private digits: number[]
  private exponent: number

  constructor() {
    this.digits = []
    this.exponent = 0
  }

  private getSummedPowers(): number {
    return this.digits.reduce((acc: number, item: number): number => acc + (item ** this.exponent), 0)
  }

  public isArmstrongNumber(number: number): boolean {
    this.digits = number
      .toString()
      .split('')
      .map(Number)

    this.exponent = this.digits.length

    return this.getSummedPowers() === number
  }
}

export default new ArmstrongNumbers()

File: solutions/csharp/armstrong-numbers/2/ArmstrongNumbers.cs
using System;
using System.Linq;

public static class ArmstrongNumbers
{
    public static bool IsArmstrongNumber(int number) {
      var chars = number.ToString().ToCharArray();
      var exponent = chars.Length;
      var sum = chars.Sum(c => Math.Pow(c - '0', exponent));
      
      return sum == number;
    }
}